ParkBlazerApp/src/pages/Reservations.tsx

```diff
@@ -0,0 +1,334 @@
+    const [showModalImage, setShowModalImage] = useState<boolean>(false);
+    const [selectedInvoice, setSelectedInvoice] = useState<Invoice | null>(null);
+    const [currentParkingSpot, setcurrentParkingspot] = useState<parkingSpace | null>(null);
+    const [selectedImage, setSelectedImage] = useState('');
+    
+    function setError(message: any) {
+        throw new Error("Function not implemented.");
+    }
+
+    useEffect(() => {
+        const fetchUserData = async () => {
+            const token = AuthService.getToken();
+            try {
+                
+                const reservationsResponse = await axios.get('https://server-y2mz.onrender.com/api/user_reservations', {
+                    headers: {
+                        Authorization: `Bearer ${token}`
+                    }
+                });
+                
+                const reservations: Reservation[] = reservationsResponse.data;
+                const allparkingspaces: parkingSpace[] = parkingspaces;
+
+                combineData(reservations, allparkingspaces);
+                
+            } catch (error: any) {
+                if (error.response && error.response.data && error.response.data.message) {
+                    setError(error.response.data.message);
+                } else {
+                    setError("An unexpected error occurred. Please try again later!");
+                }
+            }
+        };
+
+        fetchUserData();
+    }, []);
+
+    const combineData = (reservations: Reservation[], parkingspots: parkingSpace[]) => {
+        const combined = reservations.map(reservation => {
+            const parkingspot = parkingspots.find(spot => spot.parkingspot_id === reservation.private_parkingspot_id);
+            return { reservation, parkingspot: parkingspot! };
+        });
+        setCombinedData(combined);
+    };
+
+    
+    const fetchInvoiceDetails = async (reservationId: number) => {
+        const token = AuthService.getToken();
+        try {
+            const response = await axios.get(`https://server-y2mz.onrender.com/api/reservation_invoice/${reservationId}`, {
+                headers: {
+                    Authorization: `Bearer ${token}`
+                }
+            });
+            setSelectedInvoice(response.data);
+            setShowModalInvoice(true);
+        } catch (error: any) {
+            if (error.response && error.response.data && error.response.data.message) {
+                setError(error.response.data.message);
+            } else {
+                setError("An unexpected error occurred. Please try again later!");
+            }
+        }
+    };
+
+    const fetchCurrentParkingspot = (spot: parkingSpace) => {
+        setcurrentParkingspot(spot);
+        setShowModalInformation(true);
+    }
+    const fetchImage = (imageUrl: string) => {
+        setSelectedImage(imageUrl);
+        setShowModalImage(true);
+    };
```

Comment:
Die Funktionsnamen beschreiben nicht ihren nutzen. Bitte umbenennen z.B. openModalImage und openModalInformation
```suggestion
    const openModalInformation = (spot: parkingSpace) => {
        setcurrentParkingspot(spot);
        setShowModalInformation(true);
    }
    const openModalImage = (imageUrl: string) => {
        setSelectedImage(imageUrl);
        setShowModalImage(true);
    };
```